Added support for graceful shutdown in SIGTERM

Shubhrajyoti-Dey-FrosTiK opened the pull request.

Commit:
Added support for graceful shutdown in SIGTERM

akshhack commented on huey/consumer.py:
```diff
@@ -498,18 +498,22 @@ def check_worker_health(self):
-            signal.signal(signal.SIGINT, signal.default_int_handler)
-        if hasattr(signal, 'SIGHUP'):
-            signal.signal(signal.SIGHUP, self._handle_restart_signal)
+            signal.signal(signal.SIGTERM, self._handle_stop_signal)
```
Shubhrajyoti PRs need to be more explanative.. why is this needed?

Commit:
Merge branch 'master' of https://github.com/Camb-ai/cambai-huey into dev

Commit:
Changed the SIGTERM handler

coleifer committed:
readthedocs yaml shit is now required.

what the fuck, eabod rtd

coleifer committed:
Add compatibility hack for Python 3.12's broken utc stuff

coleifer committed:
Updated docs on shutdown behavior.

coleifer committed:
Allow users to override the TaskWrapper implementation.

Refs #783

coleifer committed:
Factor consumer main loop into its own method.

Refs #783

coleifer committed:
Fix bug introduced when normalizing UTC stuff for python 3.12

coleifer committed:
Clarify docs slightly on revoked behavior, clean up code.

Fixes #784

coleifer committed:
Experimental SQS/S3 storage layer.

coleifer committed:
Ensure that SIGHUP followed by SIGINT, the SIGINT is respected.

Refs #755

coleifer committed:
Add example for SIGNAL_INTERRUPTED

Replaces #797

coleifer committed:
Example of limiting the time a task can run for.

coleifer committed:
2.5.1

coleifer committed:
Short doc for `on_commit_task()` Django helper.

Refs #805

coleifer committed:
Remove junk sqs implementation and requirements for utc param in storage.

coleifer committed:
Some notes to help with testing.

coleifer committed:
Add notes on redis connection pool to greenlet doc

Fixes #809

coleifer committed:
Add new ResultTimeout subclass for when Result.get times out.

Fixes #810

coleifer committed:
Also catch TimeoutError when making blocking calls.

Refs #813

coleifer committed:
Ensure a bad task serialization does not ruin whole batch of scheduled.

Fixes #815

coleifer committed:
2.5.2

coleifer committed:
Use FOR UPDATE SKIP LOCKED when supported by the database.

Replaces #817

coleifer committed:
Github actions update.

coleifer committed:
Add support for SIGNAL_ENQUEUED.

coleifer committed:
Clarify docs on new enqueued signal.

coleifer committed:
Add add'l doc on RetryTask

[skip ci]

vspiespro committed:
Update django.rst

Prevent having a ValueError [1] error if we forgot the `name` parameter to `task` and `on_commit_task` if we decorate the same function twice.

[1] https://github.com/coleifer/huey/blob/master/huey/registry.py#L27

coleifer committed:
Merge pull request #829 from vspiespro/patch-1

Update django.rst

coleifer committed:
2.5.3

coleifer committed:
Ensure preserve param doc present.

Replaces #847

coleifer committed:
Bump ubungu for githovel runner

francoisfreitag committed:
Provide `call_local` for django on_commit_task

Executes tasks wrapped with on_commit_task, which is pretty convenient
in a testing environment.

coleifer committed:
Merge pull request #848 from francoisfreitag/on-commit-call-local

Provide `call_local` for django `on_commit_task`

tcely committed:
Add `periodic_task` to mini.rst

coleifer committed:
Merge pull request #851 from tcely/patch-1

Add `periodic_task` to mini.rst

coleifer committed:
Add limit.

Fixes #853

coleifer committed:
Add acquire/release/locked methods to TaskLock.

Replaces #858

coleifer committed:
Do not allow fallback to __len__ for truthiness of Huey instance.

Fixes #859

coleifer committed:
Python 3.14 compat for specifying "fork" multiprocessing.

Fixes #867

coleifer committed:
2.5.4

coleifer committed:
Fix for pypi being a piece of shit

cbm755 committed:
Python 3.13 and Python 3.14 are supported

Fixes #874.

cbm755 committed:
metadata for pypi to link to changelog and docs

coleifer committed:
Merge pull request #876 from cbm755/pypi_changelog_link

metadata for pypi to link to changelog and docs

coleifer committed:
Merge pull request #875 from cbm755/supported_python

Python 3.13 and Python 3.14 are supported

coleifer committed:
2.5.5

coleifer committed:
Fix outdated timezone reference.

coleifer committed:
New build system

coleifer committed:
2.6.0

coleifer committed:
Remove license until @pypa figures out how to handle 3.8

Refs https://github.com/pypa/setuptools/issues/4903

coleifer committed:
Exclude docs and examples from wheel

arnavmehta7 committed:
Merge remote-tracking branch 'upstream/master' into jan-7-26-sync

arnavmehta7 committed:
Merge pull request #4 from Camb-ai/jan-7-26-sync

Jan 7 26 sync
